### PR TITLE
Assign server response outside try block

### PR DESCRIPTION
**Pull request by bluestealth**
An attempt to fix [Sendgrid client doesn't close TCP connection properly #165](https://github.com/sendgrid/sendgrid-java/issues/165).

### CODE REVIEW

**Comment by thinkingserious** [comment]
Hello @bluestealth,

Thanks for the PR!

Could you please take a moment to [sign our CLA](https://github.com/sendgrid/java-http-client/blob/master/CONTRIBUTING.md#cla) so we can review?

File: src/main/java/com/sendgrid/Client.java [hunk 1]
```diff
@@ -277,23 +278,19 @@ public Response delete(Request request) throws URISyntaxException, IOException {
 	}
 
 	private Response executeApiCall(HttpRequestBase httpPost) throws IOException {
-		CloseableHttpResponse serverResponse = null;
-		Response response = new Response();
+		CloseableHttpResponse serverResponse = httpClient.execute(httpPost);
```
Comment (thinkingserious) [inline src/main/java/com/sendgrid/Client.java]: I don't understand how this fixes https://github.com/sendgrid/sendgrid-java/issues/165. It seems like the original code should be closing the connection at the end. Could you please elaborate on how your solution fixes the issue?

Some other questions:
1. Did you test this? If so, could you describe them? I'd like to reproduce.
2. By moving the `httpClient.execute` out of the try/catch/finally, how will we handle any exceptions at that function call.

Thanks!

Comment (bluestealth) [reply src/main/java/com/sendgrid/Client.java]: You are correct that this doesn't fix the error, and could result in some exceptional states not being caught, but I can imagine some sequence of events where serverResponse.close() isn't called with the existing code. Give me a bit to think about that.

With this finalizer as long as the client nulls their reference to Class SendGrid or the class goes out of scope the port should be released. I have tested this on my own machine by nulling my reference, forcing garbage collection and then busy waiting to keep the process open.

``` java
    @Override
    public void finalize() throws Throwable {
        try {
            this.httpClient.close();
        } catch(IOException e) {
            throw new Throwable(e.getMessage());
        } finally {
            super.finalize();
        }
    }
```

Comment (thinkingserious) [reply src/main/java/com/sendgrid/Client.java]: Thanks for the follow up!

I'm not quite sure what is going on here, but I plan to investigate further.

File: build.gradle [hunk 2]
```diff
@@ -12,7 +12,7 @@
  */
 
 apply plugin: 'java'
-apply plugin: 'fatjar'
+apply plugin: "eu.appsatori.fatjar"
```
Comment (thinkingserious) [inline build.gradle]: It looks like this plugin is no longer under development, per: https://github.com/musketyr/gradle-fatjar-plugin

Do you mind converting to https://github.com/johnrengelman/shadow?

**Review by thinkingserious — CHANGES_REQUESTED** [review]

**Review by thinkingserious — APPROVED** [review]

**Comment by thinkingserious** [comment]
@bluestealth,

Thanks again for your support! Please take a moment to fill out [this form](https://docs.google.com/forms/d/e/1FAIpQLSelP4uF-wGJebxVYq1DYw5WzHomEUauab3YiqrshyCkRs0fqw/viewform#start=invite) so that we send you some swag :)

**Pull request merged**